libs/langchain/langchain/chat_models/base.py

```diff
@@ -77,7 +106,120 @@ def init_chat_model(
             gpt_4o.invoke("what's your name")
             claude_opus.invoke("what's your name")
             gemini_15.invoke("what's your name")
+
+
+    Create a partially configurable model with no default model:
+        .. code-block:: python
+
+            # pip install langchain langchain-openai langchain-anthropic
+            from langchain.chat_models import init_chat_model
+
+            # We don't need to specify configurable=True if a model isn't specified.
+            configurable_model = init_chat_model(temperature=0)
+
+            configurable_model.invoke(
+                "what's your name",
+                config={"configurable": {"model": "gpt-4o"}}
+            )
+            # GPT-4o response
+
+            configurable_model.invoke(
+                "what's your name",
+                config={"configurable": {"model": "claude-3-5-sonnet-20240620"}}
+            )
+            # claude-3.5 sonnet response
+
+    Create a fully configurable model with a default model and a config prefix:
+        .. code-block:: python
+
+            # pip install langchain langchain-openai langchain-anthropic
+            from langchain.chat_models import init_chat_model
+
+            # We don't need to specify if config_prefix or configure_any is specified.
+            configurable_model_with_default = init_chat_model(
+                "gpt-4o",
+                model_provider="openai",
+                config_prefix="foo",
```

Comment:
When would a user want to provide a prefix?

Reply:
when they've got multiple models in a chain/graph